extensions/sql-bindings/src/test/service/azureFunctionsService.test.ts

```diff
@@ -193,9 +190,149 @@ describe('AzureFunctionsService', () => {
 			const testWatcher = TypeMoq.Mock.ofType<vscode.FileSystemWatcher>().object;
 			sinon.stub(azureFunctionUtils, 'waitForNewFunctionFile').withArgs(sinon.match.any).returns({ filePromise: Promise.resolve('TestFileCreated'), watcherDisposable: testWatcher });
 
+			should(connectionInfo.database).equal('my_db', 'ConnectionInfo database should not be changed');
 			await azureFunctionService.createAzureFunction(tableTestNode);
 
 			should(showErrorStub.calledOnce).be.true('showErrorMessage should have been called');
 		});
 	});
+
+	describe('Cancel/Error scenarios for Azure Function with SQL Binding ', function (): void {
+		it('Should prompt connection profile when user cancels selecting database', async function (): Promise<void> {
+			// This test will re-prompt the user to choose connection profile
+			sinon.stub(azureFunctionUtils, 'getAzureFunctionsExtensionApi').resolves(testUtils.azureFunctionsExtensionApi.object); // set azure functions extension api
+			sinon.stub(azureFunctionUtils, 'getAzureFunctionProject').resolves(projectFilePath); //set azure function project to have one project
+			sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);
+
+			let connectionInfo: IConnectionInfo = createTestCredentials();// Mocks promptForConnection
```

Comment:
This is just creating the connectionInfo right? Not mocking promptForConnection.